src/index.ts

```diff
@@ -0,0 +1,148 @@
+import escape from 'xml-escape';
+import * as Cypress from 'cypress';
+import RunResult = CypressCommandLine.RunResult;
+import CypressRunResult = CypressCommandLine.CypressRunResult;
+import CypressFailedRunResult = CypressCommandLine.CypressFailedRunResult;
+import BeforeRunDetails = Cypress.BeforeRunDetails;
+import PluginConfigOptions = Cypress.PluginConfigOptions;
+import PluginEvents = Cypress.PluginEvents;
+import Spec = Cypress.Spec;
+import { ConfigOption, JUnitTestSuite } from './type';
+import Reporter, { TestCase, TestSuite } from './reporter';
+
+let reporter: Reporter;
+
+function onBeforeRun(details: BeforeRunDetails) {
+  // The `projectName` is not officially documented in Cypress documentation.
+  // As used in this context, it represents the basename of the current working directory.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  reporter.junitSuite.name = `Cypress Test - ${(details.config as any)?.projectName} ${details.browser?.displayName}`;
+}
+
+function onAfterSpec(spec: Spec, results: RunResult) {
+  const testsuite = new TestSuite();
+  let errCount = 0;
+  results.tests.forEach((test) => {
+    const testcase = new TestCase(
+      // `test.title` includes both the context name (defined in 'describe' or 'context' blocks within Cypress)
+      // and the specific test name.
+      // The last part of `test.title` is the test name itself.
+      // Anything before the last part is the context name.
+      // Consequently, the JUnit testcase name is constructed by combining the context name with the test name.
+      test.title.join(' '),
+      test.title.length ? test.title[test.title.length - 1] : '',
+      msToSec(test.duration),
+      spec.relative || '',
+    );
+    if (test.state === 'failed') {
+      testcase.failure = {
+        message: escape(test.displayError || ''),
```

Comment:
Do we really need to use `escape`? I tried feeding `message` some XML that wasn't escaped and I its escaped properly by the underlying XML lib without relying on `escape`. Have I missed a use case?

Reply:
My use case is like following
```
<failure message="AssertionError: Timed out retrying after 4000ms: expected '&lt;input#email1.form-control.action-email&gt;' to have value 'fake@email.comm', but the value was 'fake@email.com'
    at Context.eval (webpack:///./cypress/e2e/examples/actions.cy.js:13:32)" type="AssertionError"><![CDATA[AssertionError: Timed out retrying after 4000ms: expected '<input#email1.form-control.action-email>' to have value 'fake@email.comm', but the value was 'fake@email.com'
    at Context.eval (webpack:///./cypress/e2e/examples/actions.cy.js:13:32)]]></failure>
```

But I haven't check if `xmlbuilder2` handles it properly. Let me try it.

Reply:
`escape` can be removed. To be honest, I decided to use `xmlbuilder2` after introducing `escape` so I wasn't aware it can handle it. And previous several candidates cannot do that. 😂